test/unified_format.py

```diff
@@ -409,8 +409,8 @@ def _operation_type(self, spec, actual, key_to_compare):
                 t for alias in spec for t in self.__type_alias_to_type(alias)])
         else:
             permissible_types = self.__type_alias_to_type(spec)
-        self.test.assertIsInstance(
-            actual[key_to_compare], permissible_types)
+        value = actual[key_to_compare] if key_to_compare else actual
+        self.test.assertIsInstance(value, permissible_types)
```

Comment:
We can assert the type of a result at the top-level in which case the key_to_compare is None.